settings.py

```diff
@@ -270,11 +270,15 @@ def dump(self, f: TextIO, level: int = 0) -> None:
             # fetch class to avoid going through getattr
             cls = self.__class__
             type_hints = cls.get_type_hints()
+            entries = [e for e in self]
+            if not type_hints and not entries:
+                # write empty dict for empty Group with no instance values
+                cls._dump_value(type_hints, f, indent="  " * level)
```

Comment:
This should then become an empty dict to be more clear
```suggestion
                cls._dump_value({}, f, indent="  " * level)
```